arch/arm/src/armv8-m/arm_cache.c

```diff
@@ -176,6 +176,78 @@ void up_disable_icache(void)
 }
 #endif
 
+/****************************************************************************
+ * Name: up_invalidate_icache
+ *
+ * Description:
+ *   Invalidate the instruction cache within the specified region.
+ *
+ * Input Parameters:
+ *   start - virtual start address of region
+ *   end   - virtual end address of region + 1
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_ARMV8M_ICACHE
+void up_invalidate_icache(uintptr_t start, uintptr_t end)
```

Comment:
@xiaoxiang781216 since the up_invalidate_icache() is basically the same for armv7-m and armv8-m, shouldn't it be possible to use a common definition for both?

Reply:
it's a little bit hard since common folder is for arm/armv6-m/armv7-a/armv7-r/armv7-m/armv8-m, but the cache operation is different from Cortex-A/R and Cortex-M.

Reply:
Ok, understood! In this case duplication is better!